Linking.md

```diff
@@ -262,30 +288,35 @@ offset of the immediate within the code section.
 The same technique applies for all function calls whether the function is
 imported or defined locally.
 
+The Wasm object files shall have imports corresponding to all undefined
+symbols, and all defined symbols have an associated export, including
+those with hidden visibility or internal (local) linkage.
+
+When taking the address of a non-existent function, a null index shall
+be written into the code/data section (for example, taking the address of
+a weak symbol which is not available at link time).
+
+XXX what about calling a weak symbol that isn't available? What can you
+do to make the resulting Wasm file valid? Bother, do we have to do something
+silly like create a stub function with the right signature that just calls
+unreachable?
+
+
 Merging Data Sections
 ---------------------
 
+Data symbols (C/C+ globals) are represented in the object file as Wasm
+data segments with an associated DATA symbol.
```

Comment:
Right now we default to one segment per symbol, but there can be many symbol per segment too (i.e. if you build with -fno-data-sections).  At least I don't think we want to mandata a 1-to-1 mapping.

Reply:
Oops, I agree, "is associated with" is a copy and paste from the previous text. But yes, the association is one-way only, DATA symbols point to their segment (which doesn't have to have one symbol only)